src/Service/QuestionTemplateService.php

```diff
@@ -52,13 +52,15 @@ public function getQuestion(int $id): QuestionTemplate
         return $question;
     }
 
-    public function getQuestions(array $filters,int $currentPage): array
+    public function getFilteredQuestionsForPage(array $filters,int $currentPage): array
     {
-        $criteria = new Criteria($filters, [], ($currentPage - 1) * 5, 5);
+        $paginator = new Paginator($this->getFilteredQuestionsNumber($filters));
+        $criteria = new Criteria($filters, [], ($currentPage - 1) * $paginator->getResultsPerPage(), $paginator->getResultsPerPage());
```

Comment:
maybe you should enable the paginator to compute limit and offset, you're using this formula in a lot of places.